sys/pylon/gstpylonsrc.c

```diff
@@ -166,6 +167,37 @@ typedef enum _GST_PYLONSRC_PROP
   PROP_TRANSFORMATION22
 } GST_PYLONSRC_PROP;
 
+typedef enum _GST_PYLONSRC_AUTOFATURE {
+  AUTOF_GAIN,
+  AUTOF_EXPOSURE,
+  AUTOF_WHITEBALANCE,
+
+  AUTOF_NUM_FEATURES,
+
+  AUTOF_NUM_MANUAL = 2
```

Comment:
This seems odd to me, won't `AUTOF_WHITEBALANCE` and `AUTOF_NUM_MANUAL` both have the value 2? Perhaps they don't belong in the same enum?

Reply:
I kind of see what you are doing here, trying to combine both auto and manual features into one enum, it works, but you gain no automatic bounds checking with the manual entries since you're defining the value of `AUTOF_NUM_MANUAL` in the enum, may as well just make it a `#define`.

Reply:
Maybe this one could be a define. But if you are concerned of auto bound checking it is already lost. Because `AUTOF_NUM_FEATURES` is part of enum but is beyond the end of related arrays. Putting `AUTOF_NUM_FEATURES` in the same enum just states that it is related to other values.

Reply:
Ok, we can leave this as is for now.